session.go

```diff
@@ -0,0 +1,160 @@
+package mockoidc
+
+import (
+	"strings"
+	"time"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// TimeFunc is the default function that the token generators will use for the current time
+var TimeFunc = time.Now
+
+// TimeNowFunc is the type of the function that can be passed tot he topken generators
+// if time.Now needs to be overriden for testing purposes
+type TimeNowFunc func() time.Time
+
+// Session is a struct that maintains session state across requests
+type Session struct {
+	SessionID  string
+	Scopes     []string
+	OAuthState string
+	OIDCNonce  string
+	User       *User
+}
+
+// SessionStore is a map of #Session structs
+type SessionStore struct {
+	Store map[string]*Session
+}
+
+// Config is the OIDCMock server configuration information (here as placeholder)
+type Config struct {
+	ClientID     string
+	ClientSecret string
+	Keypair      *Keypair
+	AccessTTL    int64
+	RefreshTTL   int64
+	Issuer       string
+}
+
+// NewSession returns a pointer to a new Session struct
+func (ss *SessionStore) NewSession(scope string, oAuthState string, oidcNonce string, user *User) (*Session, error) {
```

Comment:
We should add a `NewSessionStore` initializer. It will need to do the `make(map[string]*Session)` logic we need to initialize the `SessionStore.Store` field.

Reply:
Yep!  Cool.  Do we want to add the sessionStore tot he Mockoidc struct?  I'm not sure there's a need, but it seems to be where it might belong.  And then if we do that, should I make the MockOIDC struct the method  receiver for NewSessionStore?

Current PR does neither of those things, but glad to change it if it makes sense